src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -88,49 +89,48 @@ public ExpenseOD addExpenseInMonth(ExpenseID expenseIDAdd) throws EntityExceptio
-        this.expenseID = expenseIDEdit;
-        SessionStorage session = expenseID.getSession();
-        MonthlyStorage month = session.getMonthlyData(expenseID.getMonthID());
-        ArrayList<Expense> monthExpenseList = month.getExpenseData();
-        ArrayList<Category> categoryList = month.getCategoryData();
-
-        try {Category selectedCategory = findCategory(categoryList, expenseID.getCategory());
-            Expense selectedExpense = findExpense(monthExpenseList, expenseID.getOldExpense());
+        try {SessionStorage session = expenseIDEdit.getSession();
+            this.month = session.getMonthlyData(expenseIDEdit.getMonthID());
+            ArrayList<Expense> monthExpenseList = month.getExpenseData();
             double valueDouble = toDouble(expenseIDEdit.getValue());
+
             if (valueDouble < 0) {
                 //Expense value less than 0 fail: When a user tries to edit the expense value with a negative number.
                 ExpenseOD expenseODFailEdit = new ExpenseOD("Expense value can't be less than $0. Please try again!");
                 return expenseOB.fail(expenseODFailEdit);}
 
             if(!Objects.equals(expenseIDEdit.getName(), expenseIDEdit.getOldExpense())){
-                if(checkHaveSameNameInList(monthExpenseList, expenseID.getName())) {
-                    // Repeated name fail: When a user tries to edit the expense name to a existing expense in month.
-                    ExpenseOD expenseODFailEdit = new ExpenseOD("There is already a expense with this new name in this month.");
+                if(checkHaveSameNameInList(monthExpenseList, expenseIDEdit.getName())) {
+                    // Repeated name fail: When a user tries to edit the expense name to an existing expense in month.
+                    ExpenseOD expenseODFailEdit = new ExpenseOD("There is already a expense with this new name " +
+                            "in this month.");
                     return expenseOB.fail(expenseODFailEdit);}}
-            if (changeInRecurringInfo()) {
-                if (expenseID.getIsRecurringExpense()) {
-                    setEditInfo(expenseID, valueDouble, selectedExpense, selectedCategory);
-                    session.addRecurExpense(selectedExpense);
+
+            if (changeInRecurringInfo(expenseIDEdit)) {
+                if (expenseIDEdit.getIsRecurringExpense()) {
+                    Expense newRecurExpense = (Expense) expenseFactory.editMonthObject(setExpenseEditorInfo(expenseIDEdit));
+                    session.addRecurExpense(newRecurExpense);
                     // Success edit to new recurring expense.
-                    ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes of this recurring expense!");
+                    ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes " +
+                            "of this recurring expense!");
```

Comment:
Maybe you could add a comment explaining what exactly is happening here so it's more understandable

Something like
```java
            if (changeInRecurringInfo(expenseIDEdit)) {
                if (expenseIDEdit.getIsRecurringExpense()) {
                    // If expense is newly recurring, we have to update it inside MonthlyStorage (where it currently is)
                    // and also add it to the recurring expenses list in SessionStorage
                    Expense newRecurExpense = (Expense) expenseFactory.editMonthObject(setExpenseEditorInfo(expenseIDEdit));
                    session.addRecurExpense(newRecurExpense);
                    // Success edit to new recurring expense.
                    ExpenseOD expenseODSuccessEditRecurring = new ExpenseOD("You have updated all changes " + "of this recurring expense!");
                    return expenseOB.success(expenseODSuccessEditRecurring);
  
```